### DISCUSSION TITLE
Add comments to Bootstrapper for hinting overridable methods.

### DISCUSSION THREAD

**Discussion by GF-Huang** [canton7/Stylet#287] [Ideas]
`BootstrapperBase`, `StyletIoCBootstrapperBase` and `Bootstrapper<T>`, each has some methods overridable. But everytime I implements my `Bootstrapper`, I always forgets which one need to call `base.XXX()`, which one no need. This makes me have to look through the source code.

**Comment by canton7** [comment]
It's always safe to call the base method, so if in doubt, do that.

**Reply by GF-Huang** [reply]
Call base at the first or at the last of the override method body?
